tests/functional/test_s3.py

```diff
@@ -2148,6 +2148,36 @@ def test_checksums_included_in_expected_operations(
             assert "Content-MD5" in stub.requests[-1].headers
 
 
+@pytest.mark.parametrize(
+    "input_param, expected_header",
+    [("foo", "foo,aws-chunked"), (None, "aws-chunked")],
+)
+def test_checksum_content_encoding(input_param, expected_header):
+    environ = {}
+    op_kwargs = {
+        "Bucket": "mybucket",
+        "Key": "mykey",
+        "Body": b"foo",
+        "ChecksumAlgorithm": "sha256",
+    }
+    if input_param is not None:
+        op_kwargs["ContentEncoding"] = input_param
+    with mock.patch("os.environ", environ):
+        environ["AWS_ACCESS_KEY_ID"] = "access_key"
+        environ["AWS_SECRET_ACCESS_KEY"] = "secret_key"
+        environ["AWS_CONFIG_FILE"] = "no-exist-foo"
+        session = create_session()
+        session.config_filename = "no-exist-foo"
+        s3 = session.create_client("s3")
+        with ClientHTTPStubber(s3) as http_stubber:
+            http_stubber.add_response()
+            s3.put_object(**op_kwargs)
+            assert (
+                http_stubber.requests[-1].headers["Content-Encoding"].decode()
+                == expected_header
+            )
```

Comment:
We can probably clean this up by using existing functions instead of rewriting the client creation function.
```suggestion
    op_kwargs = {
        "Bucket": "mybucket",
        "Key": "mykey",
        "Body": b"foo",
        "ChecksumAlgorithm": "sha256",
    }
    if input_param is not None:
        op_kwargs["ContentEncoding"] = content_encoding
    s3 = _create_s3_client(
        region='us-west-2', is_secure=True, s3_config={}, signature_version='s3v4'
    )
    with ClientHTTPStubber(s3) as http_stubber:
        http_stubber.add_response()
        s3.put_object(**op_kwargs)
        request_headers =  http_stubber.requests[-1].headers
        assert request_headers["Content-Encoding"] == expected_header
```

Reply:
I thought about this but opted for consistency with the other checksum tests right above this especially since the test has nothing to with a secure request, custom config or signature version. I don't feel strongly about this though.

Reply:
Yeah, I think the existing checksum test case is the outlier rather than the rule. All other tests use `BaseS3OperationTest` or `_create_s3_client`. We should avoid continuing a bad pattern.

The args aren't super important but we can add sensible defaults to the `_create_s3_function` if needed like we're doing here. I'd say save that for a separate PR though to avoid conflating the two.

Reply:
Yeah totally I agree.